### PR TITLE
Botan autotools updates

### PR DESCRIPTION
**Pull request by dewyatt**
This does a few closely-related things:

* Moves `ax_check_botan.m4` into `m4/` with the others
* Fixes the issue where botan includes, etc are appended multiple times
* Adds a version check to make sure we're using Botan >= 2.1 (was a TODO item in configure.ac)

### CODE REVIEW

File: m4/ax_check_botan.m4 [hunk 1]
```diff
@@ -111,6 +111,23 @@ AC_DEFUN([AX_CHECK_BOTAN], [
...
+      #endif
+      ]])],
+      [
+          AC_MSG_RESULT([ok])
+          $1
+      ], [
+          AC_MSG_RESULT([failed])
+          $2
+      ]
+    )
```
Comment (zgyarmati) [inline m4/ax_check_botan.m4]: 2.2 is needed already. The current, unreleased git version of botan sets the version in build.h to 2.2.0, so we can use the same logic, just the value is different.

Comment (dewyatt) [inline m4/ax_check_botan.m4]: I had a feeling that might be the case, will update.

File: m4/ax_check_botan.m4 [hunk 2]
```diff
@@ -130,4 +149,5 @@ AC_DEFUN([AX_CHECK_BOTAN_DEFINES], [
             $2
         ],
         [#include <botan/build.h>])
+    CPPFLAGS="$save_CPPFLAGS"
```
Comment (catap) [inline m4/ax_check_botan.m4]: We can't build rnp without botan, can we? So, why we should keep `CPPFLAGS` without `BOTAN_INCLUDES`? Maybe just add it and don't touch all  `Makefile.am`?

Comment (dewyatt) [inline m4/ax_check_botan.m4]: 1. I would say it's bad behavior for an AC function to modify CPPFLAGS/etc.
2. I much prefer to keep things fine-grained and not set things like that globally. Not all of our makefile targets require botan headers.

**Comment by ronaldtse** [comment]
LGTM, thanks @dewyatt !

File: m4/ax_check_botan.m4 (hunk 2 above)
Comment (catap) [inline m4/ax_check_botan.m4]: @dewyatt, got your point.

LGTM.

**Pull request merged**